src/com/oltpbenchmark/jdbc/InstrumentedPreparedStatement.java

```diff
@@ -0,0 +1,117 @@
+package com.oltpbenchmark.jdbc;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.ParameterMetaData;
+import java.sql.PreparedStatement;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.List;
+
+import com.oltpbenchmark.api.InstrumentedSQLStmt;
+import org.HdrHistogram.ConcurrentHistogram;
+import org.HdrHistogram.Histogram;
+
+import com.oltpbenchmark.api.SQLStmt;
+
+public class InstrumentedPreparedStatement {
+
+    private final PreparedStatement stmt;
+    private final Histogram histogram;
+
+    public InstrumentedPreparedStatement(PreparedStatement stmt, InstrumentedSQLStmt sqlStmt) {
+        this.stmt = stmt;
+        this.histogram = sqlStmt.getHistogram();
+    }
+
+    public InstrumentedPreparedStatement(PreparedStatement stmt, Histogram histogram) {
+        this.stmt = stmt;
+        this.histogram = histogram;
+    }
+
+    public ResultSet executeQuery() throws SQLException {
+        long start = System.nanoTime();
```

Comment:
Can we measure the query perf using a flag in config/workload_all.xml?
That way we can disable/enable this additional latency measurement.

Reply:
added.